tests/pos/i3273/client_2.scala

```diff
@@ -0,0 +1,5 @@
+class Client {
+  val a = (_: Bar_1.A[AnyRef]).foo()
+  val b = (_: Bar_1.B[AnyRef]).bar()
+  def test(x: Bar_1.A[AnyRef]): Bar_1.B[_] = x
+}
```

Comment:
It would be good to test these cases under joint compilation, too (usually changes to classfile parser need to be mirrored in `JavaParsers`)